pkg/cmd/roachtest/cluster.go

```diff
@@ -2666,12 +2666,19 @@ func (c *clusterImpl) StopGrafana(ctx context.Context, l *logger.Logger, dumpDir
 func (c *clusterImpl) WipeForReuse(
 	ctx context.Context, l *logger.Logger, newClusterSpec spec.ClusterSpec,
 ) error {
+	if c.IsLocal() {
+		return errors.New("cluster reuse is disabled for local clusters to guarantee a clean slate for each test")
+	}
 	l.PrintfCtx(ctx, "Using existing cluster: %s (arch=%q). Wiping", c.name, c.arch)
 	if err := c.WipeE(ctx, l, false /* preserveCerts */); err != nil {
 		return err
 	}
-	if err := c.RunE(ctx, c.All(), fmt.Sprintf("rm -rf %s %s", perfArtifactsDir, goCoverArtifactsDir)); err != nil {
-		return errors.Wrapf(err, "failed to remove perf/gocover artifacts dirs")
+	// We remove the entire shared user directory between tests to ensure we aren't
+	// reusing files from previous tests, i.e. cockroach binaries, perf artifacts.
+	// N.B. we don't remove the entire home directory to safeguard against this ever
+	// running locally and deleting someone's local directory.
+	if err := c.RunE(ctx, c.All(), fmt.Sprintf("sudo rm -rf /home/%s/*", config.SharedUser)); err != nil {
```

Comment:
Why do we need `sudo`?

Reply:
Without sudo it runs into permission errors when trying to `rm` dependencies installed by go, i.e. `github.com/jstemmer/go-junit-report@latest` used by `pgx`. Which causes the cluster wipe to fail.

Reply:
Hrmm, true, but I think using `sudo` is not ideal, in the same vein of being careful in the off chance this ends up running in `local` mode. Are there a lot of tests that would need fixing? If not, I think "fixing" this in the tests with a `defer c.Run("go clean -modcache")` would be better.

I wish we  could just use `-modcacherw` in the install below, but that doesn't work currently either (https://github.com/golang/go/issues/64282).

https://github.com/cockroachdb/cockroach/blob/7bb38b5d91576f3afe29ac4cbcb29bd11caf0d4e/pkg/cmd/roachtest/tests/pgx.go#L82

Reply:
Only four so I think it's worth fixing these tests up to remove the sudo. Luckily it looks like apt-get + curl dependencies don't run into the same issue.